File: src/main/java/ListNode.java
public class ListNode {

    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }

    @Override
    public String toString() {
        StringBuilder result = new StringBuilder(val + "");
        ListNode current = next;
        while (current != null) {
            result.append("->" + current.val);
            current = current.next;
        }
        return result.toString();
    }

}

File: src/main/kotlin/ListNode.kt
class ListNode(var `val`: Int = 0) {
    var next: ListNode? = null

    override fun toString(): String {
        val result = StringBuilder(`val`.toString())
        var current: ListNode? = next
        while (current != null) {
            result.append("->${current.`val`}")
            current = current.next
        }
        return result.toString()
    }

}